### PR TITLE
Vcs zip fix

### PR DESCRIPTION
**Pull request by belsten**
Fix for version control system zip downloads. The actual submission files were not in the zip.
New zip folder should contain two top directories each containing a folder for each student. One directory should contain the students actual submissions and the other should contain submission info. 

Will need to be tested on live server due to a lack of ability to test VCS on local machine.

### CODE REVIEW

File: site/app/controllers/MiscController.php [hunk 1]
```diff
@@ -315,85 +312,183 @@ private function downloadAssignedZips() {
...
+                                  $relativePath = substr($filePath, strlen($temp_path) + 1);
+                                  // Add current file to archive
+                                  $zip->addFile($filePath, $file . "/" . $relativePath);
+                              }
+                          }
+                          $x = $arr_length; //cuts the for loop early when found
+                      }
+                  }
+              }
+           }
```
[the remark this answers is not in the document]
Comment (belsten) [reply site/app/controllers/MiscController.php]: Yes, for now. I haven't been able to test with a real VSC gradable. There are more issues that need to be fixed but this one was brought up to me by a professor who had a TA that could get submissions so I want them to be able to have some functionality until a better solution is found.

File: site/app/controllers/MiscController.php [hunk 2]
```diff
@@ -203,23 +202,21 @@ private function displayFile() {
...
 
         $this->core->getOutput()->useHeader(false);
         $this->core->getOutput()->useFooter(false);
         header('Content-Type: application/octet-stream');
-        header("Content-Transfer-Encoding: Binary"); 
-        header("Content-disposition: attachment; filename=\"{$filename}\"");
-        readfile(pathinfo($_REQUEST['path'], PATHINFO_DIRNAME) . "/" . basename(rawurldecode(htmlspecialchars_decode($_GET['path']))));
+        header("Content-Transfer-Encoding: Binary");
+        header("Content-disposition: attachment; filename=\"{$_REQUEST['file']}\"");
+        readfile(pathinfo($_REQUEST['path'], PATHINFO_DIRNAME) . "/" . rawurlencode( basename($_REQUEST['path'])));
```
[the remark this answers is not in the document]
Comment (belsten) [reply site/app/controllers/MiscController.php]: Any changes here can be ignored. I did not change those calls. My branch must have not been updated

**Comment by MasterOdin** [comment]
To help with the de-duplication, I'd suggest doing something like this:
```
$paths = ['submissions'];
if ($gradeable->useVcsCheckout()) {
    $paths[] = 'checkout';
}
$zip = new \ZipArchive();
$zip->open($zip_name, \ZipArchive::CREATE | \ZipArchive::OVERWRITE);
foreach ($paths as $path) {
    $gradeable_path = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), $path, $gradeable->getId());
    $zip->addEmptyDir($path);
    // add files to $zip in $path
    ...
}
```

**Comment by belsten** [comment]
I refactored the code; it's looking a lot better now. The new structure also downloads the checkout and submission folder when downloading a singular students VCS submissions

File: site/app/controllers/MiscController.php [hunk 3]
```diff
@@ -315,85 +315,89 @@ private function downloadAssignedZips() {
...
-                            }
-                        }
-                        $x = $arr_length; //cuts the for loop early when found 
-                    }
-                } 
-            }
+        foreach ($paths as $path) {
+          $gradeable_path = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), $path,
+              $gradeable->getId());
+          if($type === "all") {
```
Comment (MasterOdin) [inline site/app/controllers/MiscController.php]: make the indentation (4 spaces per level) consistent here please

**Pull request merged**